PVTPackage/source/MultiphaseSystem/PhaseSplitModel/DeadOilFlash.cpp

```diff
@@ -32,10 +32,12 @@ DeadOilFlash::DeadOilFlash( std::vector< std::vector< double > > const & PVDO,
                             double waterSurfaceMassDensity,
                             double waterSurfaceMolecularWeight )
   :
-  m_oilPhaseModel( pvt::PHASE_TYPE::OIL, PVDO, oilSurfaceMassDensity, oilSurfaceMolecularWeight ),
-  m_gasPhaseModel( pvt::PHASE_TYPE::GAS, PVDG, gasSurfaceMassDensity, gasSurfaceMolecularWeight ),
-  m_waterPhaseModel( PVTW, waterSurfaceMassDensity, waterSurfaceMolecularWeight )
+  m_oilPhaseModel( pvt::PHASE_TYPE::OIL, PVDO, oilSurfaceMassDensity, oilSurfaceMolecularWeight )
 {
+  DeadOil_PhaseModel * ptr1 = new DeadOil_PhaseModel( pvt::PHASE_TYPE::GAS, PVDG, gasSurfaceMassDensity, gasSurfaceMolecularWeight );
+  m_gasPhaseModel = std::unique_ptr< DeadOil_PhaseModel >( ptr1 );
+  BlackOil_WaterModel * ptr2 = new BlackOil_WaterModel( PVTW, waterSurfaceMassDensity, waterSurfaceMolecularWeight );
+  m_waterPhaseModel = std::unique_ptr< BlackOil_WaterModel >( ptr2 );      
```

Comment:
you could typically use `std::make_unique`
```suggestion
  m_gasPhaseModel = std::make_unique< DeadOil_PhaseModel >( pvt::PHASE_TYPE::GAS, PVDG, gasSurfaceMassDensity, gasSurfaceMolecularWeight );
  m_waterPhaseModel = std::make_unique< BlackOil_WaterModel >( PVTW, waterSurfaceMassDensity, waterSurfaceMolecularWeight );    
```

Reply:
Done